subsys/bluetooth/controller/ll_sw/ull_central_iso.c

```diff
@@ -534,6 +563,23 @@ uint8_t ll_cig_parameters_commit(uint8_t cig_id, uint16_t *handles)
 	ll_iso_setup.cis_idx = 0U;
 
 	return BT_HCI_ERR_SUCCESS;
+
+cleanup:
+	/* Late configuration failure - clean up */
+	for (uint8_t i = 0U; i < num_cis; i++) {
+		if (cis_created_handles[i] != LLL_HANDLE_INVALID) {
+			/* Release CIS instance created in failing configuration */
+			cis = ll_conn_iso_stream_get(cis_created_handles[i]);
+			ll_conn_iso_stream_release(cis);
+		}
```

Comment:
```suggestion
		if (cis_created_handles[i] != LLL_HANDLE_INVALID) {
			/* Release CIS instance created in failing configuration */
			cis = ll_conn_iso_stream_get(cis_created_handles[i]);
			ll_conn_iso_stream_release(cis);
		} else {
			break;
		}
```

Looks like `cis_created_handles` are allocated in a way that means we can break once we see `LLL_HANDLE_INVALID`

Reply:
The idea is that CISes could have been created earlier than this call to `ll_cig_parameters_commit`, which is why there is this array keeping track of thoses CISes created in _this_ session. For that reason, breaking would defeat the purpose.

Reply:
But `cis_created_handles` only contains handles of CIS created in this session, right? Is it not being populated by existing CIS handles, so we should be able to avoid any "hole" in this array

Reply:
Currently the code uses cig->lll.num_cis as index into the array. This will be '1' coming into the function if one was created earlier. The code may possibly be refactored to only iterate over newly created CISes - then we could break.